src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -2284,6 +2285,42 @@ private[immutable] final class HashMapBuilder[K, V] extends ReusableBuilder[(K,
       aliased
     }
 
+  def result[V1](f: V => V1): HashMap[K, V1] = {
+    if (rootNode.size == 0) {
+      HashMap.empty
+    } else if (aliased != null) {
+      aliased.transform((_, v) => f(v))
+    } else {
+      def loop(node: MapNode[K, V]): Unit = {
+        node match {
+          case bm: BitmapIndexedMapNode[K, V] =>
+            val iN = bm.payloadArity
+            val jN = bm.nodeArity
+            val content = bm.content
+            var i = 0
+            while (i < iN) {
+              val value = bm.getValue(i)
+              val newValue = f(value)
+              content(TupleLength * i + 1) = newValue
```

Comment:
this seems a bit dangerous and means that multiple calls to `builder.result` yield different things

in fact the second call to `builder.result(_.result())` would result in a class cast exception because the rootNode values are no longer of the correct type

Reply:
This is not public API (the enclosing class is `private[collection]`) but we could make this safer by making subsequent use of the builder throw an `IllegalStateException`.

I'm currently pursuing an alternative implementation which appears faster, and which doesn't require this facility so I might end up dropping this change.